Java-Majasdarbi/src/main/java/advanced/test/Exercise3/Joiner.java

```diff
@@ -0,0 +1,21 @@
+package advanced.test.Exercise3;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+public class Joiner {
+    private String separator;
+
+    public Joiner(String separator) {
+        this.separator = separator;
+    }
+
+    @SafeVarargs
+    public final <T> String join(T... items) {
+        String res = Arrays.stream(items)
+                .map(i -> i.toString() + separator)
+                .collect(Collectors.joining());
+        return res.substring(0, res.length() - separator.length());
```

Comment:
Šeit ir tāda nelielas problēmas apiešana. Paskaties ko dara, ja Collectors.joining metodei padod separator. Tad arī varēsi no lambdas `i -> i.toString() + separator` ņemt ārā to `+ separator`